server/src/main/java/org/elasticsearch/index/query/IntervalQueryBuilder.java

```diff
@@ -132,6 +134,19 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
             fieldType.indexOptions().compareTo(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS) < 0) {
             throw new IllegalArgumentException("Cannot create IntervalQuery over field [" + field + "] with no indexed positions");
         }
+        Set<String> maskedFields = new HashSet<>();
+        sourceProvider.extractFields(maskedFields);
+        for (String maskedField : maskedFields) {
+            MappedFieldType ft = context.fieldMapper(maskedField);
+            if (ft == null) {
+                return new MatchNoDocsQuery();
+            }
+            if (ft.tokenized() == false ||
+                ft.indexOptions().compareTo(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS) < 0) {
+                throw new IllegalArgumentException("Cannot create IntervalQuery over field ["
+                    + maskedField + "] with no indexed positions");
```

Comment:
Is this really needed ? Couldn't we check these options in `MappedFieldType#intervals` since `use_field`  is only applicable on the `match` rule ?

Reply:
++, we already check for this in `TextFieldMapper#intervals` so the check is unnecessary here